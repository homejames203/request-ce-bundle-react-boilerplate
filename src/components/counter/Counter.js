import React from 'react';
import DocumentTitle from 'react-document-title';

export const Counter = ({ counter, inc, dec, double }) =>
  <DocumentTitle title="Counter">
    <h2>
      <button type="button" className="decButton" onClick={dec}>-</button>
      Counter: {counter}
      <button type="button" className="incButton" onClick={inc}>+</button>
      <button type="button" className="dblButton" onClick={double}>* 2</button>
    </h2>
  </DocumentTitle>;
