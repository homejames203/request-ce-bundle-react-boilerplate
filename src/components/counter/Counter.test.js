import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Counter } from './Counter';

describe('<Counter />', () => {
  let props;
  let component;

  beforeEach(() => {
    props = {
      dec: jest.fn(),
      inc: jest.fn(),
      double: jest.fn(),
      counter: 0,
    };
    component = shallow(<Counter {...props} />);
  });

  test('verify snapshot', () => {
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  describe('decrement button', () => {
    let button;

    beforeEach(() => {
      button = component.find('.decButton').first();
    });

    test('should be rendered', () => {
      expect(button).toBeDefined();
    });

    test('should call the "dec" prop function when clicked', () => {
      button.simulate('click');

      expect(props.dec).toBeCalled();
    });

    test('should have "-" as a label', () => {
      expect(button.text()).toEqual('-');
    });
  });

  describe('increment button', () => {
    let button;

    beforeEach(() => {
      button = component.find('.incButton').first();
    });

    test('should be rendered', () => {
      expect(button).toBeDefined();
    });

    test('should call the "inc" prop function when clicked', () => {
      button.simulate('click');
      expect(props.inc).toBeCalled();
    });

    test('should have "+" as a label', () => {
      expect(button.text()).toEqual('+');
    });
  });

  describe('double button', () => {
    let button;

    beforeEach(() => {
      button = component.find('.dblButton').first();
    });

    test('should be rendered', () => {
      expect(button).toBeDefined();
    });

    test('should call the "double" prop function when clicked', () => {
      button.simulate('click');
      expect(props.double).toBeCalled();
    });

    test('should have "* 2" as a label', () => {
      expect(button.text()).toEqual('* 2');
    });
  });
});
