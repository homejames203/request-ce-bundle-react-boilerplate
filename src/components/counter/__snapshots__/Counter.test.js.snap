// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> verify snapshot 1`] = `
<SideEffect(DocumentTitle)
  title="Counter"
>
  <h2>
    <button
      className="decButton"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    Counter: 
    0
    <button
      className="incButton"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="dblButton"
      onClick={[Function]}
      type="button"
    >
      * 2
    </button>
  </h2>
</SideEffect(DocumentTitle)>
`;
